blueflood-core/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/RollupHandler.java

```diff
@@ -56,114 +63,145 @@
 
     private static final boolean ROLLUP_REPAIR = Configuration.getInstance().getBooleanProperty(CoreConfig.REPAIR_ROLLUPS_ON_READ);
     private ExecutorService ESUnitExecutor = null;
+    private ListeningExecutorService RollupsOnReadExecutor = null;
+
+    private TimeValue rollupOnReadTimeout = new TimeValue(5, TimeUnit.SECONDS); //Hard-coded for now, make configurable if need be
     
     protected RollupHandler() {
         if (Util.shouldUseESForUnits()) {
             // The number of threads getting used for ES_UNIT_THREADS, should at least be equal netty worker threads
             ESUnitExecutor = Executors.newFixedThreadPool(Configuration.getInstance().getIntegerProperty(CoreConfig.ES_UNIT_THREADS));
         }
+        RollupsOnReadExecutor = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(Configuration.getInstance().getIntegerProperty(CoreConfig.ROLLUP_ON_READ_THREADS)));
```

Comment:
`ListeningDecorator` will delegate the tasks to the internal thread pool which we pass, and returns a list of futures and then we wait for the entire list to complete. We can wait for a list of futures without really using `ListeningDecorator` which return `ListenableFuture`s.